node/plugin/tests/plugin_test.go

```diff
@@ -0,0 +1,269 @@
+	operator := testConfig.Operators[0]
+	assert.Equal(t, "0", operator.NODE_QUORUM_ID_LIST)
+
+	testConfig.RunNodePluginBinary("opt-out", operator)
+
+	tx := getTransactor(t, operator)
+	operatorID := getOperatorId(t, operator)
+
+	registeredQuorumIds, err := tx.GetRegisteredQuorumIdsForOperator(context.Background(), operatorID)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(registeredQuorumIds))
+
+	ids, err := tx.GetNumberOfRegisteredOperatorForQuorum(context.Background(), core.QuorumID(0))
+	assert.NoError(t, err)
+	assert.Equal(t, uint32(0), ids)
+
+	testConfig.RunNodePluginBinary("invalid", operator)
+
+	registeredQuorumIds, err = tx.GetRegisteredQuorumIdsForOperator(context.Background(), operatorID)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, len(registeredQuorumIds))
+
+	ids, err = tx.GetNumberOfRegisteredOperatorForQuorum(context.Background(), core.QuorumID(0))
+	assert.NoError(t, err)
+	assert.Equal(t, uint32(0), ids)
+}
+
+func getOperatorId(t *testing.T, operator deploy.OperatorVars) [32]byte {
+	_, privateKey, err := getECDSAPrivateKey(operator.NODE_ECDSA_KEY_FILE, operator.NODE_ECDSA_KEY_PASSWORD)
+	assert.NoError(t, err)
+	assert.NotNil(t, privateKey)
+
+	logger, err := logging.GetLogger(logging.DefaultCLIConfig())
+	assert.NoError(t, err)
+	assert.NotNil(t, logger)
+
+	ethConfig := geth.EthClientConfig{
+		RPCURL:           operator.NODE_CHAIN_RPC,
+		PrivateKeyString: *privateKey,
+	}
+
+	client, err := geth.NewClient(ethConfig, logger)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	transactor, err := eth.NewTransactor(logger, client, operator.NODE_BLS_OPERATOR_STATE_RETRIVER, operator.NODE_EIGENDA_SERVICE_MANAGER)
+	assert.NoError(t, err)
+	assert.NotNil(t, transactor)
+
+	kp, err := bls.ReadPrivateKeyFromFile(operator.NODE_BLS_KEY_FILE, operator.NODE_BLS_KEY_PASSWORD)
+	assert.NoError(t, err)
+	assert.NotNil(t, kp)
+
+	g1point := &core.G1Point{
+		G1Affine: kp.PubKey.G1Affine,
+	}
+	keyPair := &core.KeyPair{
+		PrivKey: kp.PrivKey,
+		PubKey:  g1point,
+	}
+
+	return keyPair.GetPubKeyG1().GetOperatorID()
+}
+
+func getTransactor(t *testing.T, operator deploy.OperatorVars) *eth.Transactor {
+	hexPk := strings.TrimPrefix(testConfig.Pks.EcdsaMap[testConfig.Deployers[0].Name].PrivateKey, "0x")
+	logger, err := logging.GetLogger(logging.DefaultCLIConfig())
+	assert.NoError(t, err)
+	assert.NotNil(t, logger)
+
+	ethConfig := geth.EthClientConfig{
+		RPCURL:           operator.NODE_CHAIN_RPC,
+		PrivateKeyString: hexPk,
+	}
+
+	client, err := geth.NewClient(ethConfig, logger)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	transactor, err := eth.NewTransactor(logger, client, operator.NODE_BLS_OPERATOR_STATE_RETRIVER, operator.NODE_EIGENDA_SERVICE_MANAGER)
+	assert.NoError(t, err)
+	assert.NotNil(t, transactor)
+
+	return transactor
+}
+
+// Returns the decrypted ECDSA private key from the given file.
+func getECDSAPrivateKey(keyFile string, password string) (*keystore.Key, *string, error) {
```

Comment:
Is this the same as this? https://github.com/Layr-Labs/eigenda/blob/d83442aef2c5663dd830bdfbf562d0ff018753fb/node/plugin/cmd/main.go#L23
If so, we can pull it out and make it sharable in tests